pkg/util/httputil/http.go

```diff
@@ -4,24 +4,49 @@
 package httputil
 
 import (
+	"crypto/tls"
 	"fmt"
+	"github.com/pingcap/tiproxy/pkg/manager/cert"
 	"io"
 	"net/http"
+	"time"
 
 	"github.com/cenkalti/backoff/v4"
 	"github.com/pingcap/tiproxy/lib/util/errors"
 	pnet "github.com/pingcap/tiproxy/pkg/proxy/net"
 )
 
-func Get(httpCli http.Client, addr, path string, b backoff.BackOff) ([]byte, error) {
+type Client struct {
+	cli          *http.Client
+	getTLSConfig func() *tls.Config
+}
+
+func NewHTTPClient(certManager *cert.CertManager) *Client {
+	if certManager == nil {
+		certManager = cert.NewCertManager()
+	}
+	return &Client{
+		cli: http.DefaultClient,
+		getTLSConfig: func() *tls.Config {
+			return certManager.ClusterTLS()
+		},
+	}
+}
+
+func (c *Client) SetTimeout(t time.Duration) {
+	c.cli.Timeout = t
```

Comment:
No, the `cli` is used concurrently. You can copy a `cli` and set the timeout.

Reply:
do you mean expose the `Cli` for other package set the timeout?

Reply:
Either is fine.
E.g. you can pass `timeout` to `Get()`, `Get()` copies `cli` and sets `timeout`.